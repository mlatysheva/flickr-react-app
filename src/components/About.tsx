import { Counter } from './Counter';
import { FakeContext } from './fakeContext';

function About() {
  return (
    <div className="main">
      <h1>Fake components as reference</h1>
      <p>I created this React application to learn React and Redux.</p>
      <p>In the application I used the mostly widely spread features that a website could have.</p>
      <Counter />
      <FakeContext />
    </div>
  );
}

export default About;
