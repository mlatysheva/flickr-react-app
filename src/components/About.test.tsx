import { render, screen } from '@testing-library/react';
import About from './About';
import { MemoryRouter } from 'react-router-dom';

test('renders text', () => {
  render(
    <MemoryRouter>
      <About />
    </MemoryRouter>);
  const text = screen.getByText(/About us/i);
  expect(text).toBeInTheDocument();
});

test('renders navigation link', () => {
  render(
    <MemoryRouter>
      <About />
    </MemoryRouter>);
  const linkElement = screen.getByText(/ABOUT/i);
  expect(linkElement).toBeInTheDocument();
});

test('renders About', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <About />
    </MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});

describe("About", () => {
  it("renders About component", () => {
    render(<MemoryRouter>
      <About />
    </MemoryRouter>);
    expect(screen.getByText(/About us/i)).toBeInTheDocument();
    expect(screen.getByRole('button')).toBeInTheDocument();
  })
})