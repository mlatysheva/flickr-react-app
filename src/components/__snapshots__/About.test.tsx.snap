// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders About 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <h1>
      About us
    </h1>
    <div>
      <h1>
        Counter: 0
      </h1>
      <button>
        +1
      </button>
    </div>
  </div>
</DocumentFragment>
`;
